tests/integration/customerc20.test.ts

```diff
@@ -219,6 +210,37 @@ const registerCustomToken = async (
     'Start l2Erc20Address not equal empty address'
   ).to.eq(constants.AddressZero)
 
+  // it should fail without the approval
+  if (isL2NetworkWithCustomFeeToken()) {
+    try {
+      const regTx = await adminErc20Bridger.registerCustomToken(
+        l1CustomToken.address,
+        l2CustomToken.address,
+        l1Signer,
+        l2Signer.provider!
+      )
+      await regTx.wait()
+      throw new Error('L2 custom token is not approved but got deployed')
+    } catch (err) {
+      expect((err as Error).message).to.contain('Insufficient allowance')
+    }
+  }
+
+  const maxFeePerGasOnL2 = (await l2Signer.provider!.getFeeData()).maxFeePerGas
+  const maxFeePerGasOnL2WithBuffer = maxFeePerGasOnL2?.mul(7)
+  // hardcode gas limit to 60k
+  const estimatedGasFee = BigNumber.from(60_000).mul(
+    maxFeePerGasOnL2WithBuffer!
+  )
+
+  if (isL2NetworkWithCustomFeeToken()) {
+    await adminErc20Bridger.approveGasTokenForCustomTokenRegistration({
+      l1Signer,
+      amount: estimatedGasFee,
+      erc20L1Address: l1CustomToken.address,
+    })
+  }
```

Comment:
for test simplicity we can just use max

```suggestion
  if (isL2NetworkWithCustomFeeToken()) {
    await adminErc20Bridger.approveGasTokenForCustomTokenRegistration({
      l1Signer,
      erc20L1Address: l1CustomToken.address,
    })
  }
```